src/bootstrap/bootstrap.py

```diff
@@ -12,9 +12,25 @@
 import sys
 import tarfile
 import tempfile
+import sqlite3
 
 from time import time
 
+def acquire_lock():
+    try:
+        con = sqlite3.Connection("xlock.db", timeout=0)
+        curs = con.cursor()
+        curs.execute("BEGIN EXCLUSIVE")
+        return curs
+    except sqlite3.OperationalError:
+        del con
+        del curs
+        print("Waiting for lock on database")
+        con = sqlite3.Connection("xlock.db", timeout=9999)
```

Comment:
sqlite3 suggests the timeout is in milliseconds -- which means this would be just 10 seconds, quite low compared to typical rustc builds. Do we have some evidence for it being in seconds or similar? Otherwise, I would suggest passing a larger value here -- probably aiming at ~1 hour or so, though it's not clear to me why we *need* a timeout -- don't we want to block indefinitely? (Can we specify that?)

Reply:
Unfortunately, in my testing, `float('inf')` doesn't work (just instantly times out), so it has to be some smaller value.
However, the sqlite3 lib for python does timeout in seconds ("The default for the timeout parameter is 5.0 (five seconds).")